src/core/src/pass/sdpa_to_paged_attention.cpp

```diff
@@ -81,9 +81,12 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
     OPENVINO_ASSERT(input_ids_node, "The model doesn't contain input_ids or input_embeds input. Aborting.");
 
     input_ids_node->set_partial_shape(PartialShape{-1});
+    auto input_ids_target_inputs = input_ids_node->get_output_target_inputs(0);
     auto unsqueezed_input_ids =
         std::make_shared<v0::Unsqueeze>(input_ids_node, v0::Constant::create(element::i32, Shape{}, {1}));
-    replace_node(input_ids_node, unsqueezed_input_ids);
+    for (const auto& target : input_ids_target_inputs) {
+        target.replace_source_output(unsqueezed_input_ids);
+    }
```

Comment:
Hey, what's going on here? `replace_node` looks to be banned everywhere. Why is in this place? And why is in this PR?

Reply:
This led to duplication of tensor names in the model.
It copies the name of the "position_ids" to the unsqueeze tensor.

Reply:
Well, it makes sense to copy the name because, in this particular place, we have two different interpretations of what we should treat as `position_ids`: the old one or the new one. Anyway, it doesn't trouble the user, because only input and output tensor names matter, everything inside is hidden. Why is it so important in this case?

Reply:
usually we do not allow to duplicate tensor names, I don't want to break this rule in this particular case